.github/workflows/e2e_tests.yaml

```diff
@@ -43,6 +43,7 @@ jobs:
           sudo rm -rf /opt/ghc
           sudo rm -rf "/usr/local/share/boost"
           sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+          sudo apt install python3.9
```

Comment:
Just wondering why it wasn't needed before?

Reply:
Python 3.8 is not supported for Ray versions past 2.10 iirc. 
We were safe before with `ray==2.7.0`.

Reply:
Also this brings on another concern in terms of future proofing.

What would we need to do to upgrade the notebook images to python 3.11? 
Seeing as 3.9 is now the oldest supported Python version for the latest Ray packages we don't want to be caught out.

Reply:
Right, it'd be good to anticipate and loop with the Notebook team.

Reply:
Python 3.9 installation in e2e tests was added in https://github.com/project-codeflare/codeflare-sdk/pull/542